src/BizHawk.Emulation.Cores/Consoles/Sega/gpgx64/GPGX.cs

```diff
@@ -26,8 +31,13 @@ public GPGX(CoreLoadParameters<GPGXSettings, GPGXSyncSettings> lp)
 
 			ServiceProvider = new BasicServiceProvider(this);
 			// this can influence some things internally (autodetect romtype, etc)
-			string romextension = "GEN";
 
+			// Determining system ID from the rom. If no rom provided, assume Genesis (Sega CD)
+			SystemId = lp.Roms.Any() ? lp.Roms[0].Game.System : VSystemID.Raw.GEN;
+
+			// We need to pass the exact file extension to GPGX for it to correctly interpret the console
+			string RomExtension = lp.Roms.Any() ? Path.GetExtension(lp.Roms[0].RomPath).Replace(".", "") : "";

```

Comment:
```suggestion
			// Determining system ID from the rom. If no rom provided, assume Genesis (Sega CD)
			SystemId = VSystemID.Raw.GEN;
			var RomExtension = string.Empty;
			if (lp.Roms.Count >= 1)
			{
				SystemId = lp.Roms[0].Game.System;
				// We need to pass the exact file extension to GPGX for it to correctly interpret the console
				RomExtension = Path.GetExtension(lp.Roms[0].RomPath).RemovePrefix('.');
			}
```
also don't use PascalCase for locals